src/zig.rs

```diff
@@ -166,6 +186,145 @@ impl zed::Extension for ZigExtension {
             .unwrap_or_default();
         Ok(Some(settings))
     }
+
+    fn dap_request_kind(
+        &mut self,
+        adapter_name: String,
+        config: serde_json::Value,
+    ) -> zed::Result<StartDebuggingRequestArgumentsRequest, String> {
+        if adapter_name != "CodeLLDB" || adapter_name != "GDB" {
+            return Err(format!("Cannot create binary for adapter: {adapter_name}"));
+        }
+
+        config
+            .get("request")
+            .and_then(|request| {
+                request.as_str().and_then(|s| match s {
+                    "launch" => Some(StartDebuggingRequestArgumentsRequest::Launch),
+                    "attach" => Some(StartDebuggingRequestArgumentsRequest::Attach),
+                    _ => None,
+                })
+            })
+            .ok_or_else(|| {
+                "Invalid request, expected 'request' to be either 'launch' or 'attach'".to_string()
```

Comment:
This is going to be an error message also when `request` is missing altogether, which is not that great.

Reply:
And tbh, is this code even being called? I believe you'd have to implement it if your extension provided some kind of debugger, and it's using an existing one right now.